impeller/entity/contents/filters/gaussian_blur_filter_contents.cc

```diff
@@ -200,6 +207,94 @@ Rect MakeReferenceUVs(const Rect& reference, const Rect& rect) {
 int ScaleBlurRadius(Scalar radius, Scalar scalar) {
   return static_cast<int>(std::round(radius * scalar));
 }
+
+class GeometryExtractor : public TypedContentsVisitor<ColorSourceContents> {
+ public:
+  void TypedVisit(ColorSourceContents* contents) override {
+    geometry_ = contents->GetGeometry();
+  }
+  std::shared_ptr<Geometry> geometry_;
+};
+
+std::shared_ptr<Geometry> GetGeometry(
+    const std::shared_ptr<FilterInput>& input) {
+  std::shared_ptr<Geometry> result;
+  std::visit(
+      [&result](auto&& arg) {
+        using T = std::decay_t<decltype(arg)>;
+        if constexpr (std::is_same_v<T, std::shared_ptr<FilterContents>>) {
+          GeometryExtractor extractor;
+          arg->Visit(&extractor);
+          result = extractor.geometry_;
+        } else if constexpr (std::is_same_v<T, std::shared_ptr<Contents>>) {
+          GeometryExtractor extractor;
+          arg->Visit(&extractor);
+          result = extractor.geometry_;
+        } else if constexpr (std::is_same_v<T, std::shared_ptr<Texture>>) {
+          // TODO (https://github.com/flutter/flutter/issues/144266): I think
+          // gaussian blurs have to work first for this to happen.
+        } else if constexpr (std::is_same_v<T, Rect>) {
+          // Does this ever happen?
+          FML_DCHECK(false) << "unimplemented";
```

Comment:
Rect inputs are sometimes used for computing coverage before we're ready to render, so we should never encounter it right now during `FilterRender`.

However, this will be useful to use later for optimizing in `GetCoverage` given that the coverage will shrink for kInner, etc.